packages/core/test/bookmark-manager.test.ts

```diff
@@ -0,0 +1,213 @@
+
+      expect(bookmarks).toEqual(neo4jBookmarks)
+    })
+
+    it('should return get bookmarks from bookmarkSupplier', () => {
+      const extraBookmarks = ['neo4j:bm03', 'neo4j:bm04']
+
+      const manager = bookmarkManager({
+        initialBookmarks: new Map([
+          ['neo4j', neo4jBookmarks],
+          ['system', systemBookmarks]
+        ]),
+        bookmarksSupplier: () => extraBookmarks
+      })
+
+      const bookmarks = manager.getBookmarks('neo4j')
+
+      expect(bookmarks).toEqual([...neo4jBookmarks, ...extraBookmarks])
+    })
+
+    it('should return call from bookmarkSupplier with correct database', () => {
+      const bookmarksSupplier = jest.fn()
+
+      const manager = bookmarkManager({
+        bookmarksSupplier
+      })
+
+      manager.getBookmarks('neo4j')
+
+      expect(bookmarksSupplier).toBeCalledWith('neo4j')
+    })
+  })
+
+  describe('getAllBookmarks()', () => {
+    it('should return all bookmarks ', () => {
+      const manager = bookmarkManager({
+        initialBookmarks: new Map([
+          ['neo4j', neo4jBookmarks],
+          ['system', systemBookmarks]
+        ])
+      })
+
+      const bookmarks = manager.getAllBookmarks()
+
+      expect(bookmarks).toEqual([...neo4jBookmarks, ...systemBookmarks])
+    })
+
+    it('should return empty if there isnt bookmarks for any db', () => {
+      const manager = bookmarkManager({})
+
+      const bookmarks = manager.getAllBookmarks()
+
+      expect(bookmarks).toEqual([])
+    })
+
+    it('should return enrich bookmarks list with supplied bookmarks', () => {
+      const extraBookmarks = ['neo4j:bmextra', 'system:bmextra', 'adb:bmextra']
+      const bookmarksSupplier = jest.fn((database?: string) => extraBookmarks)
+      const manager = bookmarkManager({
+        initialBookmarks: new Map([
+          ['neo4j', neo4jBookmarks],
+          ['system', systemBookmarks]
+        ]),
+        bookmarksSupplier
+      })
+
+      const bookmarks = manager.getAllBookmarks()
+
+      expect(bookmarks.sort()).toEqual(
+        [...neo4jBookmarks, ...systemBookmarks, ...extraBookmarks].sort()
+      )
+    })
+
+    it('should call bookmarkSupplier for getting all bookmarks', () => {
+      const bookmarksSupplier = jest.fn()
+      const manager = bookmarkManager({
+        initialBookmarks: new Map([
+          ['neo4j', neo4jBookmarks],
+          ['system', systemBookmarks]
+        ]),
+        bookmarksSupplier
+      })
+
+      manager.getAllBookmarks()
+
+      expect(bookmarksSupplier).toBeCalledWith()
+    })
+  })
+
+  describe('updateBookmarks()', () => {
+    it('should remove previous bookmarks and new bookmarks an existing db', () => {
+      const newBookmarks = ['neo4j:bm03']
+      const manager = bookmarkManager({
+        initialBookmarks: new Map([
+          ['neo4j', neo4jBookmarks],
+          ['system', systemBookmarks]
+        ])
+      })
+
+      manager.updateBookmarks(
+        'neo4j',
+        manager.getAllBookmarks(),
```

Comment:
Could be worth adding a previous bookmark that does not exist in the BMM. In the real world this could also happen if another tx has caused the BMM to remove a BM that was still there when the first tx got started.